src/combinatorics/quiver/covering.lean

```diff
@@ -0,0 +1,257 @@
+/-
+Copyright (c) 2022 Antoine Labelle, Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Labelle, Rémi Bottinelli
+-/
+import combinatorics.quiver.basic
+import combinatorics.quiver.symmetric
+import combinatorics.quiver.cast
+import data.sigma.basic
+
+/-!
+# Covering
+
+This file defines coverings of quivers as prefunctors that are bijective on the
+so-called stars and costars at each vertex of the domain.
+
+## Main definitions
+
+* `quiver.star u` is the type of all arrows with source `u`;
+* `quiver.costar u` is the type of all arrows with target `u`;
+* `prefunctor.star φ u` is the obvious function `star u → star (φ.obj u)`;
+* `prefunctor.costar φ u` is the obvious function `costar u → costar (φ.obj u)`;
+* `prefunctor.is_covering φ` means that `φ.star u` and `φ.costar u` are bijections for all `u`;
+* `quiver.star_path u` is the type of all paths with source `u`;
+* `prefunctor.star_path u` is the obvious function `star_path u → star_path (φ.obj u)`.
+
+## Main statements
+
+* `prefunctor.path_star_bijective` states that if `φ` is a covering, then `φ.star_path u` is a
+  bijection for al `u`.
+  In other words, any path in the codomain of `φ` lifts uniquely to its domain.
+
+## TODO
+
+Clean up the namespaces by renaming `prefunctor` to `quiver.prefunctor`.
+
+## Tags
+
+Cover, covering, quiver, path, lift
+-/
+
+open function quiver
+
+universes u v w
+
+variables {U : Type*} [quiver.{u+1} U]
+          {V : Type*} [quiver.{v+1} V] (φ : U ⥤q V)
+          {W : Type*} [quiver.{w+1} W] (ψ : V ⥤q W)
+
+/-- The `quiver.star` at a vertex is the collection of arrows whose source is the vertex. -/
+@[reducible] def quiver.star (u : U) := Σ (v : U), (u ⟶ v)
+
+/-- The `quiver.costar` at a vertex is the collection of arrows whose target is the vertex. -/
+@[reducible] def quiver.costar (u : U) := Σ (v : U), (v ⟶ u)
+
+/-- A prefunctor induces a map of quiver.stars at any vertex. -/
```

Comment:
Either use the fully qualified name or a human-readable one, but don't mix both.
```suggestion
/-- A prefunctor induces a map of `quiver.star` at any vertex. -/
```
```suggestion
/-- A prefunctor induces a map of the vertex stars. -/
```